Add description of `dataType` parameter (#64747)
Azure/azure-content: articles/azure-functions/functions-bindings-storage-blob-input.md

The [configuration](#configuration) section explains these properties.

> [!NOTE]

> > If you want to read binary data from the blob, make sure to specify `dataType` as `binary`. The default value for `dataType` is `string`, which means that data is interpreted as unicode.

Reviewer: Actually, the default `dataType` is [undefined](https://github.com/Azure/azure-functions-host/blob/7c81c0d416dc3402cbd3c023ca80e0778322d6ba/src/WebJobs.Script.Grpc/azure-functions-language-worker-protobuf/src/proto/FunctionRpc.proto#L361).

When DataType is undefined, it will use rich binding (e.g. `def main(input: func.InputStream)`). https://github.com/Azure/azure-functions-python-worker/tree/dev/tests/endtoend/blob_functions/blob_trigger

When DataType is 'string', it will use generic binding and cast the blob input type as string
(e.g. `def main(input: str):`). https://github.com/Azure/azure-functions-python-worker/tree/dev/tests/endtoend/blob_functions/get_blob_as_str

Same when DataType is 'binary', it will use generic binding and cast the blob to a bytes Python object
(e.g. `def main(input: bytes):`).
https://github.com/Azure/azure-functions-python-worker/tree/dev/tests/endtoend/blob_functions/get_blob_as_bytes

cc: @vrdmr @stefanushinardi

> > If you want to read binary data from the blob, make sure to specify `dataType` as `binary`. The default value for `dataType` is `string`, which means that data is interpreted as unicode.

Reviewer 2: Adjustments based on @Hazhzeng's notes.

```suggestion
The `dataType` property determines which binding is used. The following values are available to support different binding strategies.

| Binding Value | Default | Description | Example |
| --- | --- | --- | --- |
| `undefined` | Y | Uses rich binding | `def main(input: func.InputStream)` |
| `string` | N | Uses generic binding and casts the input type as a `string` | `def main(input: str)` |
| `binary` | N | Uses generic binding and casts the input blob as `bytes` Python object | `def main(input: bytes)` |
```

> > If you want to read binary data from the blob, make sure to specify `dataType` as `binary`. The default value for `dataType` is `string`, which means that data will be interpreted as unicode.

Reviewer 2: ```suggestion
> If you want to read binary data from the blob, make sure to specify `dataType` as `binary`. The default value for `dataType` is `string`, which means that data is interpreted as unicode.
```

|**name** | n/a | The name of the variable that represents the blob in function code.|
|**path** |**BlobPath** | The path to the blob. |
|**connection** |**Connection**| The name of an app setting that contains the [Storage connection string](../storage/common/storage-configure-connection-string.md) to use for this binding. If the app setting name begins with "AzureWebJobs", you can specify only the remainder of the name here. For example, if you set `connection` to "MyStorage", the Functions runtime looks for an app setting that is named "AzureWebJobsMyStorage". If you leave `connection` empty, the Functions runtime uses the default Storage connection string in the app setting that is named `AzureWebJobsStorage`.<br><br>The connection string must be for a general-purpose storage account, not a [blob-only storage account](../storage/common/storage-account-overview.md#types-of-storage-accounts).|

> |**dataType**| n/a | For dynamically typed languages, specifies the underlying data type. Possible values are `string`, `binary`, or `stream`. More details [here](functions-triggers-bindings?tabs=python#trigger-and-binding-definitions) |

Reviewer 2: ```suggestion
|**dataType**| n/a | For dynamically typed languages, specifies the underlying data type. Possible values are `string`, `binary`, or `stream`. For more more detail, refer to the [triggers and bindings concepts](functions-triggers-bindings?tabs=python#trigger-and-binding-definitions). |
```

Author: Thank you for the great updates!
